src/index.d.ts

```diff
@@ -539,6 +555,27 @@ export function statusMatrix(args: {
   pattern: string;
 }): Promise<StatusMatrix>;
 
+export function tag(args: {
+  core?: string;
+  fs?: any;
+  dir: string;
+  gitdir?: string;
+  name: string;
+  annotated?: {
+    message: string;
+    tagger?: {
+      name?: string;
+      email?: string;
+      date?: Date;
+      timestamp?: number;
+      timezoneOffset?: number;
+    };
+    signingKey?: string;
```

Comment:
We should move `signingKey` out of the `annotated` object, and leave `signature` in there. That way if users are parsing tags, sending them over JSON, and recreating them they can preserve the original `signature.

Reply:
I was overlooking that use case and feel somewhat hard to understand. I guess the new params will look:

```typescript
  annotated?: {
    message: string;
    tagger?: {
      name?: string;
      email?: string;
      date?: Date;
      timestamp?: number;
      timezoneOffset?: number;
    };
    signature?: string;
  };
  signingKey?: string;
```

and `signature` may be kind of `-----BEGIN PGP SIGNATURE-----\n...`, is this right?

Reply:
Yup! That's what I was thinking. And probably I should just throw an error if both signature and signingKey exist.

Reply:
Hmm, cannonical `git tag` requires a message if the tag will be signed, so I don't think we should move `signingKey` out of `annotated`.

https://git-scm.com/docs/git-tag
> If one of -a, -s, or -u <keyid> is passed, the command creates a tag object, and requires a tag message. Unless -m <msg> or -F <file> is given, an editor is started for the user to type in the tag message.

Reply:
Eh... the `signingKey` is a private PGP key though. I don't want that to "accidentally" get shipped around as part of some `annotated` object. The _key_ shouldn't be part of the tag object is what I'm saying.

I see what you're saying about the TypeScript type, but I think we just throw an error if `signingKey` is provided and `annotated` is not. Interestingly, this problem goes away if we make `annotatedTag` it's own function. Might be another reason to do so then.

Reply:
Okay, I understand now 😄